jwcrypto/tests.py

```diff
@@ -1497,6 +1515,31 @@ def test_decrypt_keyset(self):
         jwt.JWT(jwt=token, key=keyset, algs=jwe_algs_and_rsa1_5,
                 check_claims={'exp': 1300819380})
 
+    def test_decrypt_keyset_dup_kid(self):
+        keyset = jwk.JWKSet.from_json(json_encode(PrivateKeys))
+        # add wrong key with duplicate kid
+        key = jwk.JWK(kid='testkey', **E_A3_key)
+        keyset.add(key)
+
+        # encrypt a new JWT with kid
+        key = jwk.JWK(kid='testkey', **E_A2_key)
+        header = copy.copy(A1_header)
+        header['kid'] = 'testkey'
+        t = jwt.JWT(header, A1_claims, algs=jwe_algs_and_rsa1_5)
+        t.make_encrypted_token(key)
+        token = t.serialize()
+
+        # try to decrypt without a matching key
+        with self.assertRaises(jwt.JWTMissingKey):
+            jwt.JWT(jwt=token, key=keyset, check_claims={'exp': 1300819380})
+
+        # add right key
+        keyset.add(key)
+
```

Comment:
This will definitely now break the way it should, I may just merge and fix the test myself later for the false nergative I've seen